File: s2core/src/main/scala/org/apache/s2graph/core/tinkerpop/structure/S2Element.scala
package org.apache.s2graph.core.tinkerpop.structure

import java.util

import org.apache.tinkerpop.gremlin.structure.{Property, Graph, Element}

class S2Element extends Element {
  override def property[V](s: String, v: V): Property[V] = ???

  override def remove(): Unit = ???

  override def properties[V](strings: String*): util.Iterator[_ <: Property[V]] = ???

  override def graph(): Graph = ???

  override def label(): String = ???

  override def id(): AnyRef = ???
}

File: s2core/src/main/java/org/apache/s2graph/core/tinkerpop/structure/S2Element.java
package org.apache.s2graph.core.tinkerpop.structure;

import org.apache.tinkerpop.gremlin.structure.Element;
import org.apache.tinkerpop.gremlin.structure.Graph;
import org.apache.tinkerpop.gremlin.structure.Property;

import java.util.Iterator;

public class S2Element implements Element {
    @Override
    public Object id() {
        return null;
    }

    @Override
    public String label() {
        return null;
    }

    @Override
    public Graph graph() {
        return null;
    }

    @Override
    public <V> Property<V> property(String s, V v) {
        return null;
    }

    @Override
    public void remove() {

    }

    @Override
    public <V> Iterator<? extends Property<V>> properties(String... strings) {
        return null;
    }
}
